specs/interop/predeploys.md

```diff
@@ -816,6 +825,78 @@ The bridging of `SuperchainERC20` using the `SuperchainTokenBridge` will require
   - `sendERC20()` should emit a `SentERC20` event. `
   - `relayERC20()` should emit a `RelayedERC20` event.
 
+## DependencyManager
+
+| Constant | Value                                        |
+| -------- | -------------------------------------------- |
+| Address  | `0x4200000000000000000000000000000000000029` |
+
+The `DependencyManager` is a predeploy contract that manages the L2-side dependency set. It maintains the list of chain IDs
+that are part of the dependency set and initiates withdrawal transactions to add new dependencies.
+
+### Functions
+
+#### `addDependency`
+
+Initiates the addition of a new chain to the dependency set through a withdrawal transaction.
+
+```solidity
+function addDependency(uint256 _chainId) external;
+```
+
+- MUST only be callable by the `DEPOSITOR_ACCOUNT`
+- MUST revert if the chain ID is already in the dependency set
+- MUST revert if the chain ID is the same as the current chain
+- MUST revert if adding the chain would exceed the maximum size (255)
+- MUST initiate a withdrawal transaction to the L1 `SuperchainConfigInterop`
+
+#### `isInDependencySet`
+
+Returns whether a chain ID is in the dependency set.
+
+```solidity
+function isInDependencySet(uint256 _chainId) public view returns (bool);
+```
+
+- MUST return true if the chain ID is the current chain's ID
+- MUST return true if the chain ID is in the dependency set
+- MUST return false otherwise
+
+#### `dependencySetSize`
+
+Returns the number of chains in the dependency set.
+
+```solidity
+function dependencySetSize() external view returns (uint8);
+```
+
+#### `dependencySet`
+
+Returns the list of chain IDs in the dependency set.
+
+```solidity
+function dependencySet() external view returns (uint256[] memory);
+```
+
+### Events
+
+#### `DependencyAdded`
+
+MUST be triggered when a new dependency is added to the set.
+
+```solidity
+event DependencyAdded(uint256 indexed chainId);
+```
+
+### Invariants
```

Comment:
A mention of `ADD_DEPENDENCY_WITHDRAWWAL_GAS_LIMIT` would be good here, as its a constant and assumed to be an invariant